course05/homework01/joel/string.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <ostream>
```

Comment:
In general IOs related header files are heavy and you should include declarations (iosfwd) in header files, definitions (like ostream) in cc. This will also force you to move your class method efinition to the cc, which is better.